mantidproject/mantid
# Create added event data in isis sans gui

Implements ticket [#11351](http://trac.mantidproject.org/mantid/ticket/11351)
## For testing:

You need to make sure that you have the access to the data archive enabled.
The user file is:  MASKSANS2D_125O_4m_12mm_Brooks.txt (attached to the original ticket) and the instrument is SANS2D

Please code review.

**Test GUI logic**
1. Open the ISIS_SANS interace and go to the Add tab
2. At the histogram section select "Use binning from monitors" from the dropdown
   - Confirm that the line edit field below the dropdown box is not editable
3. At the histogram section select "Use custom binning" from the dropdown
   - Confirm that the line edit field below the dropdown box is editable
4. At the histogram section select "Save as event data" from the dropdown
   - Confirm that the line edit field below the dropdown box is not editable

**Test data format of added event data**
1. Open the ISIS_SANS interace and go to the Add tab
2. Add two workspaces to the "Run files to sum" list (27019 and 26995)
3. At the histogram section select "Save as event data" from the dropdown
4. Press the "Sum" button
   - Confirm that one and only one new file was created in your save directory
   - Confirm that it ends with -add
5. Load the file into MantidPlot
   - Confirm that the loaded workspace is a group workspace with two child workspaces
   - Confirm that the child workspace names are {GroupWSName}+"-add_added_event_data" and one with {GroupWSName}+"-add_monitors_added_event_data" (Note that the added event data tag is only used to distunguish the child workspace name from the group workspace name as they cannot be the same) 
   - Confirm that the child workspace {GroupWSName}+"-add_added_event_data" is an event workspace and that the child workspace {GroupWSName}+"-add_monitors_added_event_data" is a Workspace2D
   - Confirm that the history of both child workspaces contain a "Plus" step where the two workspaces you added are recorded

**Test size of added event data**
1. Open the ISIS_SANS interace and go to the Add tab
2. Add two workspaces to the "Run files to sum" list (27019 twice)
3. At the histogram section select "Save as event data" from the dropdown
4. Press the "Sum" button
5. Load the workspace 27019 in the ISIS_SANS interace. Make a note of the number of events
6. Load the newly added workspace in the ISIS_SANS interface.
   - Confirm that the number of events is essentially doubled

**Test for regular rebinning**
1. Open the ISIS_SANS interace and go to the Add tab
2. Add two workspaces to the "Run files to sum" list (27019 and 26995)
3. At the histogram section select "Use custom binning" from the dropdown
4. Press the "Sum" button
   - Confirm that one and only one new file was created in your save directory
   - Confirm that it ends with -add
5. Load the file into MantidPlot
   - Confirm that there is a single Workspace2D
6. Repeat these steps with  "Use binning from monitors" selected
